docs(observability): Add advanced configuration, authentication, and telemetry collection docs (#1969)
firebase/genkit: docs/plugins/firebase.md

### Telemetry

> The Firebase plugin includes an implementation for sending telemetry data,
> including metrics, traces, and logs to Firebase Genkit Monitoring. To get started,

Reviewer: ```suggestion
The Firebase plugin provides a telemetry implementation for sending metrics, traces, and logs to Firebase Genkit Monitoring. To get started,
```

Author: Done.
